cylc/uiserver/scripts/gui.py

```diff
@@ -20,8 +20,31 @@
 For a multi-user system see `cylc hub`.
 """
 
+import sys
+
+# from cylc.flow.id_cli import parse_id, parse_id_async
+from cylc.flow.exceptions import InputError
+
 from cylc.uiserver.app import CylcUIServer
 
 
 def main(*argv):
-    return CylcUIServer.launch_instance(argv or None)
+    workflow_id = None
+    for arg in argv:
+        if arg.startswith('-'):
+            continue
+        try:
+            # workflow_id, _, _ = parse_id(
+            # arg, constraint='workflows', infer_latest_runs=True)
+
+            # The following line needs to be replaced with the line above in
+            # order to interpret the workflow_id. This, causes problems with
+            # the ServerApp Event loop.
+
+            workflow_id = arg
```

Comment:
!!!
This pr currently only accepts the full workflow id e.g. ` cylc gui workflow-name/run4` rather than `cylc gui workflow-name`

This is due to a problem calling `parse_id`. The async is resulting in the ServerApp error of `There is no current event loop in thread 'MainThread'.`